### DISCUSSION TITLE
Sort bookmarks with sortspec file

### DISCUSSION THREAD

**Discussion by syrth-io** [SebastianMC/obsidian-custom-sort#153] [Q&A]
the drag-and-drop sorting via bookmarks is a cool feature, and I imagine pretty useful for some other users, but I'm wondering if the opposite is possible--sorting the bookmarks themselves with the `sortspec.md` file. it probably sounds counterintuitive, but my computer doesn't play nice with some Electron features (or Electron itself sometimes), and manually dragging-and-dropping files and bookmarks is agonizingly slow, even for a short-distance move. this is a problem, because this plugin has me mad with sorting power over my files (and also I'm neurodivergent and need things a certain way for my brain to work lmao, which this plugin has been a HUGE help with!), so trying to work with my bookmarks in comparison is absolutely painful.

I would love to be able to use my sortspec file on my bookmarks. I'm not sure if my case is too much of an edge case to warrant the addition of that feature if not already available, but I thought I'd bring it up anyway just in case! I probably could just edit the `bookmarks.json` directly, but it's not as convenient (also apologies if it is possible and I missed it--I looked through the docs, open and closed issues, and discussions and didn't see anything, but it's entirely possible I missed something anyway.)

**Comment by SebastianMC** [comment]
Hi @syrth-io 

while the idea which you bring up is tempting and I understand its value, it is far out of scope of the custom sort plugin capabilities.

From technical perspective it also poses some challenges: while the sorting spec parsing and processing can be directly reused, the actual data model and applying to Bookmarks tree and UI and wiring the events is completely different from File Explorer. Not mentioning that Bookmarks and its data model are internal to Obsidian and much more complex (it doesn't reflect the filesystem, as the File Explorer does), so it would be a pure hacking endeavour. And the resulting feature very very prone to breaking changes in subsequent Obsidian releases (which happens often).

In other words: a high effort on start to implement that feature and then guarantee of the need of verification and time-consuming updates and fixes after each Obsidian release.

At the same time I have a related idea which could be feasible, if you specify your requirements more precisely and confirm some points.
Namely, as you know, bookmarks are stored in an json file inside the vault and you can edit it manually. Instead of manual editing a command line tool can be created to reorganize the boomarks.json file.
Do you know command line tools?
Can you use them?
It would be a command line script requiring local installation of `nodejs` or `Deno` in order to run.
Is this acceptable and feasible on your side?
Then I could think about extraction of the sorting parser and sorting engine from the plugin and enabling it to run off-line, as the standalone command line tool, which modifies bookmarks.json according to sorting specification.
I have no idea of how much effort that would involve, at the same time it seems tempting, much much simpler and less risky than including such functionality directly into the existing custom sort plugin.

Let me know what you think!
